src/ReactionNetwork.jl

```diff
@@ -0,0 +1,341 @@
+"""
+Macro that inputs an expression corresponding to a reaction network and output a Reaction Network Structure that can be used as input to generation of SDE and ODE and Jump problems.
+Most arrows accepted (both right, left and bi drectional arrows).
+Note that while --> is a correct arrow, neither <-- nor <--> works.
+Using non-filled arrows (⇐, ⟽, ⇒, ⟾, ⇔, ⟺) will disable mass kinetics and lets you cutomize reaction rates yourself.
+Use 0 or ∅ for degradation/creation to/from nothing.
+Example systems:
+    rn = @reaction_network_new rType begin #Creates a reaction network of type rType.
+        2.0, X + Y --> XY                  #This will have reaction rate corresponding to 2.0*[X][Y]
+        2.0, XY ← X + Y                    #Identical to 2.0, X + Y --> XY
+        (2.0,1.0), X + Y ↔ XY              #Identical to reactions (2.0, X + Y --> XY) and (1.0, XY --> X + Y).
+        2.0, X + Y ⟾ XY                   #Ignores mass kinetics. This will have reaction rate corresponding to 2.0*[X].
+        hill(XY,2,2,2), X + Y --> XY       #Reaction inis activated by XY according to a hill function. hill(x,v,K,N).
+        mm(XY,2,2), X + Y --> XY           #Reaction inis activated by XY according to a michaelis menten function. mm(x,v,K).
+        2.0, (X,Y) --> 0                   #This corresponds to both X and Y degrading at rate 2.0.
+        (2.0, 1.0), (X,Y) --> 0            #This corresponds to X and Y degrading at rates 2.0 and 1.0, respectively.
+        2.0, (X1,Y1) --> (X2,Y2)           #X1 and Y1 becomes X2 and Y2, respectively, at rate 2.0.
+    end
+
+    kB = 2.0; kD = 1.0
+    p = [kB, kD]
+    p = []
+    rn = @reaction_network_new type begin
+        (kB, kD), X + Y --> XY            #Lets you define parameters outside on network. Parameters can be changed without recalling the network.
+    end kB, kD
+"""
+#Macro to create a reaction network model.
+macro reaction_network_new(name, ex::Expr, p...)
+    coordinate(name, ex, p)
+end
+
+#Declare various arrow types symbols used for the empty set (also 0).
+empty_set = Set{Symbol}([:∅])
+fwd_arrows = Set{Symbol}([:>, :→, :↣, :↦, :⇾, :⟶, :⟼, :⥟, :⥟, :⇀, :⇁, :⇒, :⟾])
+bwd_arrows = Set{Symbol}([:<, :←, :↢, :↤, :⇽, :⟵, :⟻, :⥚, :⥞, :↼, :↽, :⇐, :⟽])
+double_arrows = Set{Symbol}([:↔, :⟷, :⥎, :⥐, :⇄, :⇆, :⇋, :⇌, :⇔, :⟺])
+no_mass_arrows = Set{Symbol}([:⇐, :⟽, :⇒, :⟾, :⇔, :⟺])      #Using this arrows will disable the program from multiplying reaction rates with the substrate concentrations. Gives user full control of reaction rates.
+disallowed_reactants = Set{Symbol}([:du, :u, :p, :t])           #These are not allowed since they are used in "return :((du,u,p,t) -> $system)", if a variable these gets replaced with e.g. u[1], which is bad. To Do: Make change so that these can be used.
```

Comment:
ParameterizedFunctions.jl avoids this issue by picking names that won't hit user names. It appends, `internal_var___`, i.e. `internal_var___du`, with then the only issues being that the user should know to use `t` for time. You probably want to make use of something similar here to reduce the number of "sacred names" that will commonly be hit.

Reply:
@korsbo pointed this out to me as well (both one and two times). I agree it is better that way (I will update it sometime today probably).